kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/DataHandler.java

```diff
@@ -246,6 +256,74 @@ private boolean isValidMediaEntry(PKMediaConfig mediaConfig) {
+                params.put("soundMode", "2"); // sound On
+            }
+        }
+
+        if (manifestMaxDownloadTime != -1) {
+            params.put("manifestDownloadTime", Long.toString(manifestMaxDownloadTime));
+            manifestMaxDownloadTime = -1;
+        }
+        if (segmentMaxDownloadTime != -1) {
+            params.put("segmentDownloadTime", Long.toString(segmentMaxDownloadTime));
+            segmentMaxDownloadTime = -1;
+        }
+        if (totalSegmentDownloadTimeMs > 0 && totalSegmentDownloadSizeByte > 0) {
+
+            double bandwidthInByteMS = totalSegmentDownloadSizeByte / (totalSegmentDownloadTimeMs * 1.0);
+            params.put("bandwidth", String.format("%.3f", convertToKbps(bandwidthInByteMS)));
+            totalSegmentDownloadTimeMs = 0;
+            totalSegmentDownloadSizeByte = 0;
+        }
+
+        if (flavorParamsId != null) {
+            params.put("flavorParamsId", flavorParamsId); // --> in live
+        }
+
+        if (targetBuffer == -1 && player != null && player.getSettings() instanceof PlayerSettings) {
+            targetBuffer = ((PlayerSettings) player.getSettings()).getLoadControlBuffers().getMaxPlayerBufferMs() / Consts.MILLISECONDS_MULTIPLIER_FLOAT;
+        }
+        if (targetBuffer > 0) {
+            params.put("targetBuffer", targetBuffer + "");
+            if (currentBufferPosition > 0 && currentPosition > 0 && currentBufferPosition > currentPosition) {
+                double forwardBufferHealth = (((currentBufferPosition - currentPosition) / Consts.MILLISECONDS_MULTIPLIER_FLOAT) / targetBuffer);
+                params.put("forwardBufferHealth", String.format("%.3f", forwardBufferHealth));
+            }
+        }
+
+        params.put("networkConnectionType", Utils.getNetworkClass(context));
+        if (maxConnectDurationMs > 0) {
+            params.put("networkConnectionOverhead", maxConnectDurationMs / Consts.MILLISECONDS_MULTIPLIER_FLOAT + ""); // 	max dns+ssl+tcp resolving time over all video segments in sec
+            maxConnectDurationMs = -1;
+        }
+
+        playTimeSum += ViewTimer.TEN_SECONDS_IN_MS - totalBufferTimePerViewEvent;
+        params.put("playTimeSum", Float.toString(playTimeSum / Consts.MILLISECONDS_MULTIPLIER_FLOAT));
+        params.put("actualBitrate", Long.toString(actualBitrate / KB_MULTIPLIER));
+        long averageBitrate = averageBitrateCounter.getAverageBitrate(playTimeSum + totalBufferTimePerEntry);
+        params.put("averageBitrate", Long.toString(averageBitrate / KB_MULTIPLIER));
+        if (currentAudioLanguage != null) {
+            params.put("audioLanguage", currentAudioLanguage);
+        }
+        if (currentCaptionLanguage != null) {
+            params.put("captionsLanguage", currentCaptionLanguage);
+        }
+    }
+
+    private double convertToKbps(double bandwidthInByteMS) {
+        bandwidthInByteMS *= 8;    // b/ms
+        bandwidthInByteMS /= 1024; // kb/ms
+        bandwidthInByteMS *= 1000; // kb/s
```

Comment:
return bandwidthInByteMS * 8 // bytes to bits
     / 1024 //  bits to kbits
     * 1000; // msec to sec

IMHO makes it clearer.

Reply:
thx donwe